sklearn/multiclass.py

```diff
@@ -1025,6 +1025,31 @@ def fit(self, X, y):
 
         return self
 
+    def predict_proba(self, X):
+        """Return faux probability estimates.
+
+        Estimates are based upon euclidean distance between the underlying
+        estimators' predictions and the class code.
+
+        Parameters
+        ----------
+        X : (sparse) array-like of shape (n_samples, n_features)
+            Data.
+
+        Returns
+        -------
+        T : (sparse) array-like of shape (n_samples, n_classes)
+            Returns the score of the sample for each class in the model,
+            where classes are ordered as they are in `self.classes_`.
+        """
+        check_is_fitted(self)
+        Y = np.array([_predict_binary(e, X) for e in self.estimators_]).T
+        dist = euclidean_distances(Y, self.code_book_)
+        # Inverse distance weighting:
+        eps = 1e-16
+        proba = (1.0 / (dist + eps)) / np.sum(1.0 / (dist + eps), axis=1)[:, np.newaxis]
+        return proba
```

Comment:
Do you have any scientific literature linked to this approach to express probability. I would have naively think to use a softmax instead of a linear mapping.

Reply:
I assume that we could invert the distance because we should have a upper bound of the distance since we deal with binary code (I did not look at the algorithm yet).

Reply:
Turning distance into a probability-like value is basically spatial interpolation. IDW was described in a classical ['A two-dimensional interpolation function for irregularly-spaced data'](https://dl.acm.org/doi/10.1145/800186.810616), among other methods.

Softmax is an option, of course, I cosidered a selectable method but an extra `predict_proba()` argument looked like a bad idea to me.

Practically: softmax decays somewhat faster, but that's about it. Both ways yield non-negative scores which sum to 1. Any method is able to come slightly ahead of the other depending on data in terms of ROC_AUC, AP and calibration curve (the avergage difference in metrics is within 0.01).